hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/login/AbstractHawkbitLoginUI.java

```diff
@@ -398,15 +441,50 @@ private void login(final String tentant, final String user, final String passwor
             }
 
         } catch (final CredentialsExpiredException e) {
-            LOGGER.debug("Credential expired", e);
+            LOG.debug("Credential expired", e);
             loginCredentialsExpiredNotification();
         } catch (final AuthenticationException e) {
-            LOGGER.debug("Authentication failed", e);
+            LOG.debug("Authentication failed", e);
             /* if not successful */
             loginAuthenticationFailedNotification();
         } catch (final Exception e) {
-            LOGGER.debug("Login failed", e);
+            LOG.debug("Login failed", e);
             loginAuthenticationFailedNotification();
         }
     }
+
+    protected Map<String, String> getParams() {
+        return params;
+    }
+
+    protected TextField getUsername() {
+        return username;
+    }
+
+    protected TextField getTenant() {
+        return tenant;
+    }
+
+    protected PasswordField getPassword() {
+        return password;
+    }
+
+    protected VaadinMessageSource getI18n() {
+        return i18n;
+    }
+
+    private static Map<String, String> getQueryParams(final URL url) {
```

Comment:
you could also use spring utilities for parsing query params, e.g. from an URL or directly from a HttpServletRequest: `        MultiValueMap<String,String> queryParams = UriComponentsBuilder.fromHttpRequest(httpRequest).build().getQueryParams();`